### PR TITLE
Optional clip

### PR DESCRIPTION
**Pull request by tomwyr**
Currently it's not possible to clip only one side of the chart. Either you need to clip everything or not clip at all. 
E.g. I'd like to hide my chart line below left labels area but prevent dot at value 0 from being partially invisible at the same time:

chart

This PR replaces `clipToBorder` with `clipData` where clipping of each side can be enabled/disabled.

### CODE REVIEW

**Comment by imaNNeoFighT** [comment]
Hi, there was a problem with `master`, that's why this branch is failed.
I've just fixed the problem, please update your branch with master,
Thanks.

File: lib/src/chart/base/axis_chart/axis_chart_data.dart [hunk 1]
```diff
@@ -35,8 +35,8 @@ abstract class AxisChartData extends BaseChartData with EquatableMixin {
     double maxX,
     double minY,
     double maxY,
-    bool clipToBorder,
     Color backgroundColor,
+    FlClipData clipData,
```
Comment (imaNNeoFighT) [inline lib/src/chart/base/axis_chart/axis_chart_data.dart:39]: Please just replace it with previous,
I mean don't change the order.
Thanks!

File: lib/src/chart/base/axis_chart/axis_chart_data.dart [hunk 2]
```diff
@@ -46,8 +46,8 @@ abstract class AxisChartData extends BaseChartData with EquatableMixin {
         maxX = maxX,
         minY = minY,
         maxY = maxY,
-        clipToBorder = clipToBorder ?? false,
         backgroundColor = backgroundColor,
+        clipData = clipData ?? FlClipData.none(),
```
Comment (imaNNeoFighT) [inline lib/src/chart/base/axis_chart/axis_chart_data.dart:50]: And here,
please prevent reordering parameters.

**Comment by imaNNeoFighT** [comment]
Excellent,
It is the most straight forward PR I've ever had.
Well done.

**Comment by imaNNeoFighT** [comment]
And please change target to the `dev` branch,
because we merge new features into the `dev` branch, then we bump up the version and ship it into the master with a new version code and tag.

**Comment by imaNNeoFighT** [comment]
Thanks for your awesome PR.
Hope to see more from you!

**Comment by tomwyr** [comment]
Thanks for quick review and this awesome package.
Cheers!

**Pull request merged**